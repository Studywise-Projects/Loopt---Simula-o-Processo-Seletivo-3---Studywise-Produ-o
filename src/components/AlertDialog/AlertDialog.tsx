import {
  Button as ButtonMui,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from '@mui/material';
import { useState } from 'react';
import Button from '../Button/Button';

interface IAlertDialog {
  buttonText: string;
  dialogTitle: string;
  dialogText: string;
  disagreeButtonText: string;
  agreeButtonText: string;
  handleClickDisagree: any;
  handleClickAgree: any;
}

function AlertDialog({
  buttonText,
  dialogTitle,
  dialogText,
  disagreeButtonText,
  agreeButtonText,
  handleClickDisagree,
  handleClickAgree,
}: IAlertDialog) {
  const [open, setOpen] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div>
      <Button
        data-testid='open-dialog-button'
        text={buttonText}
        handleClick={handleClickOpen}
        isAuxButton={true}
      />
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{dialogTitle}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            {dialogText}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <ButtonMui
            data-testid='disagree-button-alert-dialog'
            onClick={() => {
              handleClickDisagree();
              handleClose();
            }}
          >
            {disagreeButtonText}
          </ButtonMui>
          <ButtonMui
            data-testid='agree-button-alert-dialog'
            onClick={() => {
              handleClickAgree();
              handleClose();
            }}
            autoFocus
          >
            {agreeButtonText}
          </ButtonMui>
        </DialogActions>
      </Dialog>
    </div>
  );
}

export default AlertDialog;
