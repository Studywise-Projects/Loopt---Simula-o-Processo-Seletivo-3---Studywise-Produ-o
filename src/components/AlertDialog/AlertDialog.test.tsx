import { render } from '@testing-library/react';
import AlertDialog from './AlertDialog';

describe('AlertDialog', () => {
  it('Should match the snapshot', () => {
    const { getByText } = render(
      <AlertDialog
        buttonText='Open Dialog'
        dialogTitle='title'
        dialogText='body'
        disagreeButtonText='disagree'
        agreeButtonText='agree'
        handleClickDisagree={undefined}
        handleClickAgree={undefined}
      />,
    );
    const element = getByText('Open Dialog');

    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByText } = render(
      <AlertDialog
        buttonText='Open Dialog'
        dialogTitle='title'
        dialogText='body'
        disagreeButtonText='disagree'
        agreeButtonText='agree'
        handleClickDisagree={undefined}
        handleClickAgree={undefined}
      />,
    );

    const element = getByText('Open Dialog');
    expect(element).toBeInTheDocument();
  });
});
