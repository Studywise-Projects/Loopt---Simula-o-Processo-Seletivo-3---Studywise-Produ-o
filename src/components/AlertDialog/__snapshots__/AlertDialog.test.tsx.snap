// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog Should match the snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
  data-test-id="button"
  tabindex="0"
  type="button"
>
  Open Dialog
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
